funcx_endpoint/funcx_endpoint/executors/high_throughput/interchange.py

```diff
@@ -587,6 +597,26 @@ def _command_server(self, kill_event):
                 logger.debug("[COMMAND] is alive")
                 continue
 
+    def enqueue_task_cancel(self, task_id):
+        """ Cancel a task on the interchange """
+        logger.debug(f"Received task_cancel request for Task:{task_id}")
+
+        found_task = False
+        for manager in self._ready_manager_queue:
+            for task_type in self._ready_manager_queue[manager]['tasks']:
+                for tid in self._ready_manager_queue[manager]['tasks'][task_type]:
+                    if tid == task_id:
+                        logger.debug(f"Task:{task_id} is running, moving task_cancel message onto queue")
+                        self.task_cancel_running_queue.put((manager, task_id))
+                        found_task = True
+                        break
+
+        if found_task is False:
+            logger.debug(f"Task:{task_id} is pending, moving task_cancel message onto trap")
```

Comment:
is it definitely `pending` or could it be in other states? (eg completed?)

Reply:
It could be, but we need not care. I think what we care about is making sure that the task does not execute and waste compute. Although, if the task is completed, and we add it to the trap, it might end up costing some wasted checks on every subsequent task. The task would never get evicted from the trap.